dppl/_memory.pyx

```diff
@@ -68,16 +68,36 @@ cdef class Memory:
 
     property _context:
         def __get__(self):
-            return self.context
+            return self.queue.get_sycl_context()
+
+    property _queue:
+        def __get__(self):
+            return self.queue
 
     def __repr__(self):
         return "<Intel(R) USM allocated memory block of {} bytes at {}>" \
             .format(self.nbytes, hex(<object>(<Py_ssize_t>self.memory_ptr)))
 
-    def _usm_type(self):
+    def _usm_type(self, context=None):
         cdef const char* kind
-        kind = DPPLUSM_GetPointerType(self.memory_ptr,
-                                      self.context.get_context_ref())
+        cdef SyclContext ctx
+        cdef SyclQueue q
+        if context is None:
+            ctx = self._context
+            kind = DPPLUSM_GetPointerType(self.memory_ptr,
+                                          ctx.get_context_ref())
+        elif isinstance(context, SyclContext):
+            ctx = <SyclContext>(context)
+            kind = DPPLUSM_GetPointerType(self.memory_ptr,
+                                          ctx.get_context_ref())
+        elif isinstance(context, SyclQueue):
```

Comment:
What if name the variable `context_or_queue` ?

Reply:
Yes, makes sense. Since the argument can be either context or queue. If we do not like long names, let us just call it `syclobj` or something other than context.